pandas/tests/frame/methods/test_replace.py

```diff
@@ -974,6 +974,8 @@ def test_replace_for_new_dtypes(self, datetime_frame):
                     }
                 ),
             ),
+            # GH 35376
+            (DataFrame([[1, 1.0], [2, 2.0]]), 1.0, 5, DataFrame([[5, 5.0], [2, 2.0]]),),
```

Comment:
can you add the .replace(1, 5) case and maybe also .replace(1.0, 5.0) and .replace(1, 5.0)